lib/reverse_string.rb

```diff
@@ -1,5 +1,30 @@
 def reverse_characters(str)
+
+    i = 0    
+    n = str.size - 1
+    ansstr = " "*n
```

Comment:
I originally had ansstr as an empty string " " but the code failed. After a few adjustments, including changing it to an empty string of size n, it worked and was satisfied. At first I thought strings were like arrays when being asked to store a value in a spot that is beyond its length, nil values would fill the gap. Since that wasn't going to happen with a string I just created a blank one.

I know after doing a quick lap around the internet that there is the << or .insert options that could achieve the creation for my answer by changing the code accordingly.